src/components/frontend-engine/frontend-engine.tsx

```diff
@@ -31,32 +32,40 @@ const FrontendEngineInner = forwardRef<IFrontendEngineRef, IFrontendEngineProps>
 		defaultValues: defaultValues,
 		resolver: yupResolver(validationSchema),
 	});
-	const { watch, handleSubmit: reactFormHookSubmit, formState } = formMethods;
-
-	// TODO: Remove logging
-	useEffect(() => {
-		const subscription = watch((value, { name, type }) => console.log(value, name, type));
-
-		return () => subscription.unsubscribe();
-	}, [watch]);
+	const { watch, handleSubmit: reactFormHookSubmit, getValues } = formMethods;
 
 	// =============================================================================
 	// HELPER FUNCTIONS
 	// =============================================================================
 	useImperativeHandle<Partial<IFrontendEngineRef>, Partial<IFrontendEngineRef>>(ref, () => ({
```

Comment:
note: reworked the handles here to fit oneservice use cases
- the original formState depends on `validationMode` and `revalidationMode` to get updated, e.g. if validationMode is onSubmit, it will only get updated on submission which is not reliable
- hence adding `isValid` method to check for validity of form at any time
- added `getValues` in case there is a need to retrieve entered values at any time